hw_i2c/sample-implementations/cypress_modus_toolbox/sensirion_hw_i2c_implementation.c

```diff
@@ -0,0 +1,231 @@
+                            CY_SCB_I2C_MASTER_ARB_LOST | CY_SCB_I2C_MASTER_ABORT_START | \
+                            CY_SCB_I2C_MASTER_BUS_ERR)
+
+/*******************************************************************************
+* Forward declaration
+*******************************************************************************/
+void mI2C_Interrupt(void);
+
+
+void mI2C_Interrupt(void)
+{
+    Cy_SCB_I2C_Interrupt(mI2C_HW, &mI2C_context);
+}
+
+/* INSTRUCTIONS
+ * ============
+ *
+ * Implement all functions where they are marked as IMPLEMENT.
+ * Follow the function specification in the comments.
+ */
+
+/**
+ * Initialize all hard- and software components that are needed for the I2C
+ * communication.
+ */
+void sensirion_i2c_init()
+{
+	cy_en_scb_i2c_status_t initStatus;
+	cy_en_sysint_status_t sysStatus;
+	cy_stc_sysint_t mI2C_SCB_IRQ_cfg =
+	{
+			/*.intrSrc =*/ scb_3_interrupt_IRQn,
+			/*.intrPriority =*/ 2UL
+	};
+	/* Initialize and enable the I2C in master mode. */
+	initStatus = Cy_SCB_I2C_Init(mI2C_HW, &mI2C_config, &mI2C_context);
+	if(initStatus != CY_SCB_I2C_SUCCESS)
+	{
+		// return STATUS_FAIL;
+	}
+
+	/* Hook interrupt service routine. */
+	sysStatus = Cy_SysInt_Init(&mI2C_SCB_IRQ_cfg, &mI2C_Interrupt);
+	if(sysStatus != CY_SYSINT_SUCCESS)
+	{
+		// return STATUS_FAIL;
+	}
+
+	/* Enable interrupt in NVIC. */
+	NVIC_EnableIRQ((IRQn_Type) mI2C_SCB_IRQ_cfg.intrSrc);
+
+	/* Enable I2C master hardware. */
+	Cy_SCB_I2C_Enable(mI2C_HW);
+	// return STATUS_OK;
+}
+
+/**
+ * Execute one read transaction on the I2C bus, reading a given number of bytes.
+ * If the device does not acknowledge the read command, an error shall be
+ * returned.
+ *
+ * @param address 7-bit I2C address to read from
+ * @param data    pointer to the buffer where the data is to be stored
+ * @param count   number of bytes to read from I2C and store in the buffer
+ * @returns 0 on success, error code otherwise
+ */
+int8_t sensirion_i2c_read(uint8_t address, uint8_t* data, uint16_t count)
+{
+	uint8_t status = STATUS_FAIL;
+	cy_en_scb_i2c_status_t errorStatus;
+	uint32_t masterStatus;
+	/* Timeout 1 sec (one unit is us) */
+	uint32_t timeout = 1000000UL;
+
+	/* Setup transfer specific parameters */
+	cy_stc_scb_i2c_master_xfer_config_t transferConfig =
+	{
+		.slaveAddress = address,
+		.buffer       = NULL,
+		.bufferSize   = 0U,
+		.xferPending  = false
+	};
+	transferConfig.buffer     = data;
+	transferConfig.bufferSize = count;
+
+	/* Initiate read transaction */
+	errorStatus = Cy_SCB_I2C_MasterRead(mI2C_HW, &transferConfig, &mI2C_context);
+	if(errorStatus == CY_SCB_I2C_SUCCESS)
+	{
+		/* Wait until master complete read transfer or time out has occured */
+		do
+		{
+			masterStatus  = Cy_SCB_I2C_MasterGetStatus(mI2C_HW, &mI2C_context);
+			Cy_SysLib_DelayUs(CY_SCB_WAIT_1_UNIT);
+			timeout--;
+
+		} while ((0UL != (masterStatus & CY_SCB_I2C_MASTER_BUSY)) && (timeout > 0));
+
+	   if (timeout <= 0)
+		{
```

Comment:
The whitespace looks a bit odd in the file. We recently added clang-format support to the repositories, you may use that to format the files. See https://github.com/Sensirion/embedded-sht/blob/master/.clang-format